src/Interpreters/QueryNormalizer.cpp

```diff
@@ -122,6 +122,16 @@ void QueryNormalizer::visit(ASTIdentifier & node, ASTPtr & ast, Data & data)
     }
 }
 
+void QueryNormalizer::visit(ASTQueryParameter & node, const ASTPtr & ast, Data & data)
+{
+    auto it_alias = data.aliases.find(node.name);
+    if (it_alias != data.aliases.end())
+        throw Exception(ErrorCodes::BAD_ARGUMENTS, "Same alias used multiple times {} ", backQuote(node.name));
```

Comment:
This check is for cases where same name is used for multiple parameters.
`CREATE VIEW test_a AS SELECT * FROM Catalog1 WHERE Price={price:UInt64} AND Quantity={price:UInt64}`
Since we are visiting the query parts & storing alias in data.aliases, I thought this would be a good place to add this check. please let me know your suggestions.

Reply:
addressed it, I have added a new variable for query parameters, its only used for checking if there are duplicate names.